nomos-blend/scheduling/src/membership.rs

```diff
@@ -30,39 +34,44 @@ pub struct Node<Id> {
 
 impl<NodeId> Membership<NodeId>
 where
-    NodeId: Clone,
+    NodeId: Clone + Hash + Eq,
 {
     #[must_use]
     pub fn new(nodes: &[Node<NodeId>], local_public_key: Option<&Ed25519PublicKey>) -> Self {
+        let mut core_nodes = HashMap::with_capacity(nodes.len());
+        let mut remote_core_nodes = Vec::with_capacity(nodes.len());
+        for node in nodes {
+            core_nodes.insert(node.id.clone(), node.clone());
+            if !matches!(local_public_key, Some(key) if node.public_key == *key) {
```

Comment:
Mhh, can't use store the local node in a simple `Node` instance instead? This way we are duplicating stuff, but I don't see any reason to do that?

Reply:
It is probably acceptable, but it was not clear what was the decision behind it. If using an indexmap is not too much work, perhaps we could use that. Otherwise we can keep this, but perhaps explain with a comment why we do what we do :)